external-crates/move/crates/move-package/src/lib.rs

```diff
@@ -256,10 +257,92 @@ impl BuildConfig {
+                "Cannot build Move model for package with address renamings"
+            ));
+        }
         let _mutx = PackageLock::lock(); // held until function returns
         model_builder::build(resolved_graph, writer)
     }
 
+    // NOTE: If there are no renamings, then the root package has the global resolution of all named
+    // addresses in the package graph in scope. So we can simply grab all of the source files
+    // across all packages and build the Move model from that.
+    // TODO: In the future we will need a better way to do this to support renaming in packages
+    // where we want to support building a Move model.
+    // This deterministically selects unique address names for every named address in the package.
+    // It returns the original address assignment to the caller.
+    pub fn move_model_for_package_with_derived_addresses<W: Write>(
+        self,
+        path: &Path,
+        writer: &mut W,
+    ) -> Result<(source_model::Model, BTreeMap<Symbol, AccountAddress>)> {
+        // resolution graph diagnostics are only needed for CLI commands so ignore them by passing a
+        // vector as the writer
+        let mut resolved_graph = self.resolution_graph_for_package(path, None, writer)?;
+        // We do not support models over packages that contain address renamings
+        if resolved_graph.contains_renaming().is_some() {
+            return Err(anyhow!(
+                "Cannot build Move model for package with address renamings"
+            ));
+        }
+
+        let root_pkg = resolved_graph
+            .package_table
+            .get_mut(&resolved_graph.root_package())
+            .unwrap();
+        let previous_assignment = root_pkg.resolved_table.clone();
+        let in_use_addrs = previous_assignment
+            .values()
+            .cloned()
+            .collect::<BTreeSet<_>>();
+        // Assign a unique address to each named address in the package deterministically.
+        // So start at 42 and increment until we find an address that is not in use (this should be
+        // always immediate in expectation).
+        let mut i = 42;
+        for (_, old_addr) in root_pkg.resolved_table.iter_mut() {
+            // If the address is already set to a non-zero address, don't change it as there may be
+            // modules that hardcode this address.
+            if *old_addr == AccountAddress::ZERO {
+                loop {
+                    let random_addr = AccountAddress::from_suffix(i);
+                    i += 1;
+                    if !in_use_addrs.contains(&random_addr) {
+                        *old_addr = random_addr;
+                        break;
+                    }
+                }
+            }
+        }
+
+        let new_address_assignment = root_pkg.resolved_table.clone();
+
+        // NB:
+        // 1. Since the model currently does not support address renamings, we don't need to worry
+        //    about tracking renamings here (yet).
```

Comment:
That's not really true, it is jus that this function doesn't support them? 
The model uses the compiler which supports them. Everything is built just fine. The output though needs to take a canonical view, so it takes the name mapping from the root package.

Reply:
I was going off of the comment (/I know the previous model did not support this). 

If the new model supports this, I can take a look at adding support for it (it shouldn't be _too_ bad although it does make things a smidge more complicated).

Reply:
I don't really understand. For clarity, the model doesn't support or not support anything--in the sense that the model just consumes the compiler output. Then after that, it needs some way of referring to packages by name, so it takes the view of the root package. I'm not really sure what else you would or could do.